### PR TITLE
Implement bit_count() distributed function and handle hexadecimal string literals in buildReturnedColumn().

### PR DESCRIPTION
**Pull request by tntnatbry**
Fixes working_tpch1/qa_fe_cnxFunctions/bug4703.sql.

### CODE REVIEW

**Comment by tntnatbry** [comment]
I will add a test case for bit_count() in the regression suite, please don't merge yet.

File: utils/funcexp/func_bitwise.cpp [hunk 1]
```diff
@@ -359,5 +359,47 @@ int64_t Func_bitxor::getIntVal(Row& row,
...
+        return 0;
+    }
+
+    uint64_t val = 0;
+
+    if (!getUIntValFromParm(row, parm[0], val, isNull, fTimeZone))
+    {
+        std::ostringstream oss;
+        oss << "bit_count: datatype of " << execplan::colDataTypeToString(operationColType.colDataType);
+        throw logging::IDBExcept(oss.str(), ERR_DATATYPE_NOT_SUPPORT);
```
Comment (drrtuy) [inline utils/funcexp/func_bitwise.cpp:389]: I wonder if this throw finishes the query?

File: utils/funcexp/func_bitwise.cpp [hunk 2]
```diff
@@ -359,5 +359,47 @@ int64_t Func_bitxor::getIntVal(Row& row,
...
+    }
+
+    int64_t count = 0;
+
+    while ( val )
+    {
+        val &= val - 1;
```
Comment (drrtuy) [inline utils/funcexp/func_bitwise.cpp]: I wonder if the Hacker's delight reciep on 'counting 1-bits in a word' is better?

File: utils/funcexp/func_bitwise.cpp (hunk 1 above)
Comment (tntnatbry) [reply utils/funcexp/func_bitwise.cpp:389]: The query is killed in case an exception is thrown here.

File: utils/funcexp/func_bitwise.cpp (hunk 2 above)
Comment (tntnatbry) [reply utils/funcexp/func_bitwise.cpp]: Hacker's Delight algo has much better performance compared to the original algo used, so the PR is updated with a new implementation.

**Review by LinuxJedi — APPROVED** [review]

**Pull request merged**